presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -151,6 +163,14 @@ public Page getOutput()
             }
         }
 
+        if (joinType == JoinType.PROBE_LOOKUP_OUTER && finishing && probe == null) {
+            while (buildSideOuterJoinCurrentPosition()) {
```

Comment:
I'd put this loop into a new method.  Then I would inline the both of the method and rework in to a simple structure.